gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/GenerateMetricSchemaTask.java

```diff
@@ -40,8 +42,16 @@
 
 @CacheableTask
 public abstract class GenerateMetricSchemaTask extends DefaultTask {
-    private final Property<String> libraryName =
-            getProject().getObjects().property(String.class).value(defaultLibraryName());
+    private final Property<String> libraryName = getProject()
+            .getObjects()
+            .property(String.class)
+            .convention(getProject().provider(this::defaultLibraryName));
+
+    @org.gradle.api.tasks.Optional
+    private final Property<String> libraryVersion = getProject()
+            .getObjects()
+            .property(String.class)
+            .convention(getProject().provider(this::defaultLibraryVersion));
```

Comment:
This will require module recompilation any time versions change, but I think the upsides are worth it. What do you think, @CRogers?